--- a/tools/perf/tests/workloads/code_with_type.c
+++ b/tools/perf/tests/workloads/code_with_type.c
@@ -1,12 +1,13 @@
 // SPDX-License-Identifier: GPL-2.0
 #include <pthread.h>
+#include <stdint.h>
 #include <stdlib.h>
 #include <signal.h>
 #include <unistd.h>
 #include <linux/compiler.h>
 #include "../tests.h"
 
-extern void test_rs(uint count);
+extern void test_rs(uint32_t count);
 
 static volatile sig_atomic_t done;
 
