diff --git a/package/devel/perf/Makefile b/package/devel/perf/Makefile
index 1bce803..809b27e 100644
--- a/package/devel/perf/Makefile
+++ b/package/devel/perf/Makefile
@@ -27,7 +27,7 @@ include $(INCLUDE_DIR)/nls.mk
 define Package/perf
   SECTION:=devel
   CATEGORY:=Development
-  DEPENDS:= +libelf +libdw +PACKAGE_libunwind:libunwind +libpthread +librt +objdump @!IN_SDK @KERNEL_PERF_EVENTS \
+  DEPENDS:= +libelf +libdw +libzstd +PACKAGE_libunwind:libunwind +libpthread +librt +objdump @!IN_SDK @KERNEL_PERF_EVENTS \
 	    +PACKAGE_libbfd:libbfd +PACKAGE_libopcodes:libopcodes +libtraceevent
   TITLE:=Linux performance monitoring tool
   VERSION:=$(LINUX_VERSION)-r$(PKG_RELEASE)
@@ -40,11 +40,13 @@ endef
 
 HOST_CFLAGS += -I$(LINUX_DIR)/tools/include
 
-TARGET_LDFLAGS += $(INTL_LDFLAGS)
+TARGET_LDFLAGS += $(INTL_LDFLAGS) -fuse-ld=bfd
 
 MAKE_FLAGS = \
 	ARCH="$(LINUX_KARCH)" \
 	NO_LIBPERL=1 \
+	NO_LIBLLVM=1 \
+	NO_RUST=1 \
 	NO_LIBPYTHON=1 \
 	NO_NEWT=1 \
 	NO_LZMA=1 \
@@ -55,7 +57,6 @@ MAKE_FLAGS = \
 	NO_LIBAUDIT=1 \
 	NO_LIBCRYPTO=1 \
 	NO_LIBUNWIND=1 \
-	NO_LIBZSTD=1 \
 	NO_LIBCAP=1 \
 	NO_JEVENTS=1 \
 	NO_SHELLCHECK=1 \
