diff --git a/net/speedtest-go/Makefile b/net/speedtest-go/Makefile
index d2a5bde..668ff01 100644
--- a/net/speedtest-go/Makefile
+++ b/net/speedtest-go/Makefile
@@ -21,6 +21,7 @@ PKG_BUILD_PARALLEL:=1
 PKG_BUILD_FLAGS:=no-mips16
 
 GO_PKG:=github.com/showwin/speedtest-go
+GO_PKG_BUILD_PKG:=github.com/showwin/speedtest-go
 
 include $(INCLUDE_DIR)/package.mk
 include ../../lang/golang/golang-package.mk
